Automattic/newspack-plugin
# Improve generated product settings

### All Submissions:






### Changes proposed in this Pull Request:



This PR improves the generated donation/subscriptions products by automatically setting some flags that we probably want on:
- Hides the donation products from the shop
- Makes the donation products virtual products (digital products), so they won't get charged shipping and have no weight/dimensions.
- Makes it so that a customer can only have max one of each donation/subscription in their cart.

This PR isn't super urgent, and can definitely wait until after the impending milestone for review.

### How to test the changes in this Pull Request:

1. Delete any donation products you have (make sure you delete them from the Trash also). We only set these settings when the product is first created, so if people go in and change them, we're not undoing all their work.
2. Go to the Donation wizard and set up your donations.
3. Go to `example.com/shop`. Observe the donation products aren't there.
4. In WP Admin, go to Products > All Products and view one of the Monthly/Yearly/Once products. There should be no quantity input, just an Add to Cart button. If you click Add to Cart multiple times, you should be prevented from adding more of the product to your cart.

### Other information:
